block/src/async_io.rs

```diff
@@ -25,6 +27,12 @@ pub trait DiskFile: Send {
     fn topology(&mut self) -> DiskTopology {
         DiskTopology::default()
     }
+    /// Returns the file descriptor of the underlying disk image file.
```

Comment:
It would be useful to document what's safe to do with this file descriptor. Am I allowed to close it, read/write from it (setting the file pointer)? Probably not.

Instead of leaking the file descriptor, is it possible to offer a higher-level interface here (`lock_underlying_file()`) and push the implementation of locking into the each implementation of the `DiskFile` trait? With a bit of library code, this would be a one-liner just as it is now for implementing `fd()`, but it would prevent having to explain what you can and cannot do with the file descriptor.

Reply:
> Instead of leaking the file descriptor, is it possible to offer a higher-level interface here (lock_underlying_file())

I can look again into it; I agree that this is a nicer design. But IIRC, I tried that and complexity got out of hand

Reply:
Ah yeah, I stick to the existing design. The new approach makes the code very ugly with little benefit. I add a safety comment instead on how to use the FDs.

The `DiskIo` trait as implementation detail of the virtio_block module anyway